### ISSUE TITLE
Session dates are not parsed correctly

### ISSUE THREAD

**Issue by ebrahimebrahim** [OpenwaterHealth/opw_neuromod_sw#4]
In the example data, which I am guessing was generated by the matlab software, the dates in `subjects/example_subject/sessions/example_session/example_session.json` are formatted like this:

```txt
{
  ...
  "date": "09-Apr-2024 11:27:30",
  ...
  "date_modified": "09-Apr-2024 11:27:30"
}
```

The python `openlifu.db.Session` attempts to parse these as some ISO 8601 format:
https://github.com/OpenwaterHealth/OpenLIFU-python/blob/0efbc2c907b00e9eb2b7265204a9373d5aefaeef/src/openlifu/db/session.py#L60-L63
(which fails)

So either
- `openlifu.db.Session` should change its parsing to `datetime.strptime(date_string, "%d-%b-%Y %H:%M:%S")`,
- or if we do want dates to be stored in an ISO format then this issue should be transferred over to the matlab code repo and we should ensure that dates are written out in ISO format

**Comment by peterhollender** [comment]
Yes, we can store them in ISO format.  Matlab handles them both equally on load, so I can just change the saving code:
```
>> datetime("2024-06-01 12:45:00")
ans = 
  datetime
   01-Jun-2024 12:45:00
>> datetime("01-Jun-2024 12:45:00")
ans = 
  datetime
   01-Jun-2024 12:45:00
```

**Comment by ebrahimebrahim** [comment]
@peterhollender I have transferred this issue to the matlab software repo, since you said we'll keep the iso format and change the matlab code to output that format
